src/game/Pet.cpp

```diff
@@ -1101,148 +1076,165 @@ bool Pet::InitStatsForLevel(uint32 petlevel, Unit* owner)
-                }
             }
-            else                                            // not exist in DB, use some default fake data
-            {
-                sLog.outErrorDb("Summoned pet (Entry: %u) not have pet stats data in DB", cinfo->Entry);
 
-                // remove elite bonuses included in DB values
-                SetCreateHealth(uint32(((float(cinfo->MaxLevelHealth) / cinfo->MaxLevel) / (1 + 2 * cinfo->Rank)) * petlevel));
-                SetCreateMana(uint32(((float(cinfo->MaxLevelMana)   / cinfo->MaxLevel) / (1 + 2 * cinfo->Rank)) * petlevel));
+            else            // does not exist in pet_levelstats DB
+            {
+                sLog.outErrorDb("warlock pet levelstats missing in DB");
 
-                SetCreateStat(STAT_STRENGTH, 22);
-                SetCreateStat(STAT_AGILITY, 22);
-                SetCreateStat(STAT_STAMINA, 25);
-                SetCreateStat(STAT_INTELLECT, 28);
-                SetCreateStat(STAT_SPIRIT, 27);
+                SetCreateStat(STAT_STRENGTH, (5 + (1 * petlevel)));
+                SetCreateStat(STAT_AGILITY, (5 + (1 * petlevel)));
+                SetCreateStat(STAT_STAMINA, (7 + (1 * petlevel)));
+                SetCreateStat(STAT_INTELLECT, (10 + (1 * petlevel)));
+                SetCreateStat(STAT_SPIRIT, (9 + (1 * petlevel)));
             }
+
             break;
         }
+
         case HUNTER_PET:
         {
-            SetUInt32Value(UNIT_FIELD_PETNEXTLEVELEXP, sObjectMgr.GetXPForPetLevel(petlevel));
-            // these formula may not be correct; however, it is designed to be close to what it should be
-            // this makes dps 0.5 of pets level
-            SetBaseWeaponDamage(BASE_ATTACK, MINDAMAGE, float(petlevel - (petlevel / 4)));
-            // damage range is then petlevel / 2
-            SetBaseWeaponDamage(BASE_ATTACK, MAXDAMAGE, float(petlevel + (petlevel / 4)));
+            SetByteValue(UNIT_FIELD_BYTES_0, 1, CLASS_WARRIOR);
+            SetSheath(SHEATH_STATE_MELEE);
+            SetByteValue(UNIT_FIELD_BYTES_2, 1, UNIT_BYTE2_FLAG_UNK3 | UNIT_BYTE2_FLAG_AURAS | UNIT_BYTE2_FLAG_UNK5);
+
+            // this enables popup window (pet abandon, cancel), original value set in CreateBaseAtCreature
+            SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_PVP_ATTACKABLE | UNIT_FLAG_RESTING);
+
+            if (petlevel < 60)
+                SetUInt32Value(UNIT_FIELD_PETNEXTLEVELEXP, sObjectMgr.GetXPForPetLevel(petlevel));
+
+            else
+            {
+                SetUInt32Value(UNIT_FIELD_PETEXPERIENCE, 0);
+                SetUInt32Value(UNIT_FIELD_PETNEXTLEVELEXP, 1000);
+            }
+
+            SetBaseWeaponDamage(BASE_ATTACK, MINDAMAGE, ((GetAttackTime(BASE_ATTACK) * petlevel) - ((GetAttackTime(BASE_ATTACK) * petlevel) / 4)) / 2000);
+            SetBaseWeaponDamage(BASE_ATTACK, MAXDAMAGE, ((GetAttackTime(BASE_ATTACK) * petlevel) + ((GetAttackTime(BASE_ATTACK) * petlevel) / 4)) / 2000);
```

Comment:
Good! I have the DB attack speed values for many beasts waiting for that change for years.